platform/admin/core/admin-core-impl/src/main/java/org/codice/ddf/admin/core/impl/SystemPropertiesAdmin.java

```diff
@@ -315,25 +312,24 @@ private SystemPropertyDetails getSystemPropertyDetails(
     return new SystemPropertyDetailsImpl(title, description, options, key, property);
   }
 
+  private void clearPropertyValue(String key, Properties systemDotProperties) {
+    systemDotProperties.put(key, "");
+  }
+
   private void updateProperty(
       String key, Map<String, String> updatedProperties, Properties systemDotProperties) {
     if (updatedProperties.containsKey(key)) {
       String value = updatedProperties.get(key);
-      systemDotProperties.put(key, value);
-      System.setProperty(key, value);
-    }
-  }
 
-  private void updatePortProperty(Properties systemDotProperties) {
-    String protocol = SystemBaseUrl.EXTERNAL.getProtocol();
+      /* Clears out the property value before setting it
+       * We have to do this because when we read in the properties, the values are expanded
+       * which can lead to a state where the value is erroneously not updated after being checked
+       * */
+      clearPropertyValue(key, systemDotProperties);
```

Comment:
❓To avoid the potential problem where a property expansion would result in an empty string, should we just remove the key from the map instead and then add it back? Would that work?

Reply:
Yes, it looks like there is a remove so I could possibly do that instead.

Reply:
What do you mean by potential problem? Do you have an example?

I don't think we can just remove values from `systemDotProperties` because the spot in the layout would also be removed:
https://github.com/apache/felix/blob/c8ed1ff4c8c954e15a80e3ddf07e1e06b88f22f0/utils/src/main/java/org/apache/felix/utils/properties/Properties.java#L394-L401

Reply:
Thanks for checking @emmberk. I was afraid of that :-(

The potential problem, although unlikely, is if a property is defined as `${var}` and `var == ""`, then this code wouldn't work.

One option would be to check if the value is already `""` and if so, set it to something else. Or use a value that we know will never exist.

Reply:
What do you mean by "wouldn't work?

Reply:
```
property=${val}
val=
```

Let's say we don't fill in property in the installer, it checks it, it's blank, so it remains $val.
Then we set val to "someString", then property = "someString", right?

Reply:
I don't think this is an issue, regardless of whether it will be rare or not.

The installer always overwrites _custom.system.properties even_ if the input box for that property wasn't changed in the installer. That means, if `property` is one of the special properties that we can set in the installer and _custom.system.properties_ originally contains `property=${val}` and `val=`, the input box for `property` will be blank in the installer. If the input box for `property` is not updated in the installer, _custom.system.properties_ will be updated to contain `property=` instead of `property=${val}`. This is what I expect.